circle.yml

```diff
@@ -0,0 +1,52 @@
+# https://discuss.circleci.com/t/python-pip-dependencies-are-not-cached-reinstalled-every-time/6965
+machine:
+  pre:
+    - sudo -H pip install --upgrade virtualenv
```

Comment:
Can't this be moved to the `dependencies` phase using `pip install --user --upgrade virtualenv`? Or is it because `~/.local/bin` is not in the path?